egs/wsj/s5/utils/lang/make_lexicon_fst_silprob.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python3
+ # Copyright   2018  Johns Hopkins University (author: Daniel Povey)
+#             2018  Jiedan Zhu
+# Apache 2.0.
+ # see get_args() below for usage message.
+import argparse
+import os
+import sys
+import math
+ # The use of latin-1 encoding does not preclude reading utf-8.  latin-1
+# encoding means "treat words as sequences of bytes", and it is compatible
+# with utf-8 encoding as well as other encodings such as gbk, as long as the
+# spaces are also spaces in ascii (which we check).  It is basically how we
+# emulate the behavior of python before python3.
+sys.stdout = open(1, 'w', encoding='latin-1', closefd=False)
+sys.stderr = open(2, 'w', encoding='latin-1', closefd=False)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description="""This script creates the
+       text form of a lexicon FST, to be compiled by fstcompile using the
+       appropriate symbol tables (phones.txt and words.txt) .  It will mostly
+       be invoked indirectly via utils/prepare_lang.sh.  The output goes to
+       the stdout.""")
+     parser.add_argument('--sil-phone', dest='sil_phone', type=str,
+                        help="""Text form of optional-silence phone, e.g. 'SIL'.  See also
+                        the --silprob option.""")
+     parser.add_argument('--sil-disambig', dest='sil_disambig', type=str,
+                        help="""Disambiguation symbol to disambiguate silence, e.g. #5.
+                        Will only be supplied if you are creating the version of L.fst
+                        with disambiguation symbols, intended for use with cyclic G.fst.
+                        This symbol was introduced to fix a rather obscure source of
+                        nondeterminism of CLG.fst, that has to do with reordering of
+                        disambiguation symbols and phone symbols.""")
+     parser.add_argument('lexiconp', type=str,
+                        help="""Filename of lexicon with pronunciation probabilities
+                        (normally lexiconp.txt), with lines of the form
+                        'word prob correction-term-for-sil correction-term-for-no-sil p1 p2...',
+                        e.g. 'a   1.0  0.3  0.6  ay'""")
+     parser.add_argument('silprobs', type=str,
+                        help="""Filename with silence probabilities, with lines of the form
+                        '<s> p(sil-after|<s>) \n
+                        </s>_s correction-term-for-sil-for-</s> \n
+                        </s>_n correction-term-for-no-sil-for-</s> \n
+                        overall p(overall-sil) \n',
+                        e.g. 'a   1.0  0.3  0.6  ay'""")
+     args = parser.parse_args()
+     return args
+
+
+def read_silprobs(filename):
+    ans = []
+    silbeginprob = -1
+    silendcorrection = -1
+    nonsilendcorrection = -1
+    siloverallprob = -1
+     with open(filename, 'r', encoding='latin-1') as f:
```

Comment:
the indentation seems to change on this line.  Check that you are not using tabs.  I'm surprised; I would expect this would cause python to barf.